pkg/hnsw/eucqueue.go

```diff
@@ -75,6 +79,13 @@ func (pq *MinQueue) Push(x any) {
 	*pq = append(*pq, item)
 }
 
+func (pq *MinQueue) Peek() *Item {
+	if len(*pq) == 0 {
+		return nil
+	}
+	return (*pq)[0]
+}
```

Comment:
Since the base type is `[]*Item` and this operation doesn't mutate the array itself, it would be better to make this a value func:

```suggestion
func (pq MinQueue) Peek() *Item {
	if len(pq) == 0 {
		return nil
	}
	return pq[0]
}
```